addons/hr_expense/models/hr_expense.py

```diff
@@ -204,11 +198,19 @@ def _convert_to_tax_base_line_dict(self, base_line=None, currency=None, price_un
             extra_context={'force_price_include': True},
         )
 
-    @api.depends('currency_rate', 'total_amount', 'tax_ids', 'product_id', 'employee_id.user_id.partner_id', 'quantity')
+    @api.depends('date', 'currency_id', 'company_currency_id', 'total_amount', 'tax_ids', 'product_id', 'employee_id.user_id.partner_id', 'quantity')
     def _compute_total_amount_company(self):
+        date_today = fields.Date.context_today(self.env.user)
         for expense in self:
+            target_currency = expense.currency_id or self.env.company.currency_id
+            currency_rate = expense.company_id and self.env['res.currency']._get_conversion_rate(
+                from_currency=target_currency,
+                to_currency=expense.company_currency_id,
+                company=expense.company_id,
+                date=expense.date or date_today,
+            )
```

Comment:
no need to get the conversion rate if it's the company currency

```suggestion
            currency_rate = expense.company_id
                                        and expense.currency_id != self.env.company.currency_id
                                        and self.env['res.currency']._get_conversion_rate(
                                            from_currency=expense.currency_id,
                                            to_currency=expense.company_currency_id,
                                            company=expense.company_id,
                                            date=expense.date or date_today,
                                        )
                                        or 1.0
```

or use an `if` if you prefer